### PR TITLE
Add Bomberman 64

### PR DESCRIPTION
**Pull request by delcake**
Adds support for Bomberman 64 challenges.

Most challenges center around clearing stages with a certain number of gold cards which are themselves hidden in the level or the product of some various challenges. Optional support for the secret world is provided in a config toggle, which will add it to objectives designed for stages that require deeper progression in the game to unlock.

### CODE REVIEW

**Comment by delcake** [comment]
I may have made a mess of the PR somehow with the prior commits being included. Probably what I get for just bashing through in Visual Studio Code lmao

**Comment by soopercool101** [comment]
You're merging into the wrong branch, you want to merge into [SerpentAI:kk](https://github.com/SerpentAI/Archipelago/tree/kk), not [SerpentAI:main](https://github.com/SerpentAI/Archipelago/tree/main), that's why unintended commits got added

File: worlds/keymasters_keep/games/bomberman_64_game.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+            "Rainbow Palace Stage 2 - VS Spellmaker",
+            "Rainbow Palace Stage 3 - Doom Castle",
+            "Rainbow Palace Stage 4 - The Final Battle!",
+        ]
+    
+    def deep_stages(self) -> List[str]:
+        deep_stages: List[str] = self.late_stages[:]
+
+        if self.archipelago_options.bomberman_64_allow_rainbow_palace:
+            deep_stages.append(self.secret_stages)
```
Comment (nbrochu) [inline worlds/keymasters_keep/games/bomberman_64_game.py:158]: This needed to be extend, not append

**Comment by nbrochu** [comment]
Added manually here: https://github.com/SerpentAI/Archipelago/commit/175cffc6851e629a39d2bb09746da23725b5f9ed

(To circumvent git issues)

**Pull request closed without merging**